pkg/client/client_venconn.go

```diff
@@ -0,0 +1,176 @@
+package client
```

Comment:
This is mostly a copy-paste from the file [`client_oauth.go`](https://github.com/jetstack/jetstack-secure/blob/473277b69510d75734a1a714cf09d47583ba53e2/pkg/client/client_oauth.go). Since both use one form or another of the OAuth flow, I think it would be good to merge both later on.